contrib/epee/include/misc_language.h

```diff
@@ -128,6 +128,59 @@ namespace misc_utils
 
   }
 
+  template<typename T>
+  T factorial(T n, T s = 0)
+  {
+    static_assert(std::is_integral<T>::value, "Integral type required.");
+
+    if (s > n || n < 0 || s < 0)
```

Comment:
`static_assert(!std::is_signed<T>::value, ...);` ?

Reply:
My idea was to support all integral types, and just error out if negative numbers are passed in. I suppose in professional grade code you expect very strict use of types, so maybe a stricter interface is appropriate here.